discovery-infra/bootstrap_in_place.py

```diff
@@ -151,6 +153,37 @@ def gather_sosreport_data(node):
     node.download_file("/tmp/sosreport.tar.bz2", IBIP_DIR)
 
 
+# noinspection PyBroadException
+def log_collection(controller, vm_ip):
+    etype, _value, _tb = sys.exc_info()
+
+    logging.info(f"Collecting logs after a {('failed', 'successful')[etype is None]} installation")
```

Comment:
Didn't know this syntax

Reply:
i'm actually abusing the fact that bool gets converted to 0/1 and using that as an index into the tuple.... it's very ugly but very readable